src/calliope/backend/expression_parser.py

```diff
@@ -821,10 +825,10 @@ def as_math_string(self) -> str:  # noqa: D102, override
         evaluated = self.as_array()
         self.eval_attrs["references"].add(self.name)
 
-        if evaluated.attrs["obj_type"] != "string":
+        if "math_repr" in evaluated.attrs:
             data_var_string = evaluated.attrs["math_repr"]
         else:
-            data_var_string = rf"\text{{{self.name}}}"
+            data_var_string = self.name
```

Comment:
dimensions have no math representation and we don't want them to be `\text{...}` as we need their plain string values to do fun things with them in the helper functions (like get their associated iterator from the math).